src/opossum_lib/cli.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2023 TNG Technology Consulting GmbH <https://www.tngtech.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+import logging
+import sys
+from pathlib import Path
+
+import click
+from spdx.model.document import Document
+from spdx.parser.error import SPDXParsingError
+from spdx.parser.parse_anything import parse_file
+
+from opossum_lib.file_generation import generate_json_file_from_tree, write_dict_to_file
+from opossum_lib.graph_generation import generate_graph_from_spdx
+from opossum_lib.tree_generation import generate_tree_from_graph
+
+
+@click.command()
+@click.option(
+    "--infile",
+    "-i",
+    help="The file containing the document to be converted.",
```

Comment:
You can add `,type=click.Path()` to get some additional features.
I think there is even `click.Path(exists=True)` that validates that the path specified really exists.
I think you can even do `click.Path(exists=True, convert=pathlib.Path)` to get it automatically converted to a Pathlib path.

Reply:
Thanks for the pointer. I added `click.Path(exists=True)`, the parse method from `tools-python` exepects a `str` is input, not a `Pathlib.path`.